### PR TITLE
Fix Catch issue with TicketController and upgrades sonarsoftware/customer_portal_framework version

### PR DESCRIPTION
**Pull request by zulandar**
This PR fixes issue #4 by clearing cache on a try/catch with pulling the ticket number and display a invalid ID error.

### CODE REVIEW

File: app/Http/Controllers/TicketController.php [hunk 1]
```diff
@@ -36,13 +36,19 @@ public function show($id)
...
-                //Clear the cache here, because you may see a ticket with ISP responses but the list may not show it yet
+            try {
+                if ($ticket->getTicketID() == $id) {
+                    $accountTicketController = new AccountTicketController();
+                    $replies = array_reverse($accountTicketController->getReplies($ticket, 1));
+                    //Clear the cache here, because you may see a ticket with ISP responses but the list may not show it yet
+                    $this->clearTicketCache();
+                    return view("pages.tickets.show", compact('replies', 'ticket'));
+                }
+            }catch (Exception $e) {
```
Comment (geneccx) [inline app/Http/Controllers/TicketController.php]: 1. Try to be more specific with the type of exception being caught. A generic catch on the `Exception` type would inadvertently swallow errors that we don't necessarily want to.
2. I'm not sure if clearing the cache and redirecting them to the index is better than say, clearing the cache and retrying to fetch the ticket once or something.
3. There should be a space before `catch`

File: app/Http/Controllers/TicketController.php [hunk 2]
```diff
@@ -36,13 +36,19 @@ public function show($id)
...
+                Log::error($e->getMessage());
```
Comment (geneccx) [inline app/Http/Controllers/TicketController.php]: You can actually log the entire exception with `report($e)`. Just logging the error does not yield very good debugging information, but maybe it's okay... @SonarSoftware ?

Comment (zulandar) [reply app/Http/Controllers/TicketController.php]: The `$e->getMessage()` was following the error handling standard already built in the app. I do agree with debugging information and don't mind changing it.

Comment (geneccx) [reply app/Http/Controllers/TicketController.php]: Yeah, it'll probably need to be addressed throughout the rest of the application as well. Could probably set this up as a separate PR.

File: app/Http/Controllers/TicketController.php (hunk 1 above)
Comment (SonarSoftware) [reply app/Http/Controllers/TicketController.php]: There's no point refetching the ticket if it's not found, it'll just fail again. We should show an error that the ticket couldn't be found and clear the cache though.

Comment (zulandar) [reply app/Http/Controllers/TicketController.php]: I've made the recommended changes.

File: composer.json [hunk 3]
```diff
@@ -13,7 +13,7 @@
         "predis/predis": "^1.0",
         "laravelcollective/html": "^5.5",
         "proengsoft/laravel-jsvalidation": "^1.6.0-dev",
-        "sonarsoftware/customer_portal_framework": ">=1.0.11",
```
Comment (geneccx) [inline composer.json]: This needs to be 1.0.12

File: composer.json [hunk 4]
```diff
@@ -13,7 +13,7 @@
...
+        "sonarsoftware/customer_portal_framework": "^1.0",
```
Comment (geneccx) [inline composer.json]: This needs to be 1.0.12

Comment (geneccx) [reply composer.json]: You'll also need to update the lockfile.

Comment (zulandar) [reply composer.json]: @geneccx lockfile is already set to 1.0.12. I've made the changes within the composer.json file.

**Pull request merged**